core/remotes/docker/resolver.go

```diff
@@ -662,9 +662,13 @@ func (r *request) retryRequest(ctx context.Context, responses []*http.Response)
 		}
 	case http.StatusRequestTimeout, http.StatusTooManyRequests:
 		return true, nil
+	case http.StatusServiceUnavailable, http.StatusGatewayTimeout, http.StatusInternalServerError:
+		// Do not retry if there are other servers to try or the same error was seen in the last request
```

Comment:
The comment is slightly confusing at first look because here we are checking if we *should* retry but the comment implies it is checking if it should not retry.